cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -32,7 +32,8 @@ The following policy provides the minimum privileges necessary for Cluster Autos
         "autoscaling:DescribeAutoScalingInstances",
         "autoscaling:DescribeLaunchConfigurations",
         "autoscaling:SetDesiredCapacity",
-        "autoscaling:TerminateInstanceInAutoScalingGroup"
+        "autoscaling:TerminateInstanceInAutoScalingGroup",
+        "ec2:DescribeInstanceTypes"
```

Comment:
OK, so like I said on #4436, I very much support this approach. I think it's much simpler and cleaner than the existing pricing API-based strategy.

The only thing that is going to be a bit tricky is this part here which requires a new IAM policy for the running cluster-autoscaler binary. This new required IAM privilege isn't onerous per-se, but what it does mean is that we can no longer pre-generate instance type information using an anonymous user (something that was previously possible since we were basically just curl'ing a public HTTPS endpoint for pricing JSON files).

I am OK with this limitation, but we should point out (maybe in the [gen.go](https://github.com/kubernetes/autoscaler/blob/01a32390344755ba4339f5bb72b8c6f2b2ae1966/cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go) file code comments?) that the user running the static instance types generator will need to be a non-anonymous IAM user with privileges to call the DescribeInstanceTypes EC2 API.

Reply:
Understood, thank you for the comments and suggestions, Jay! I'll add the requested documentation shortly.